qt/scientific_interfaces/ISISReflectometry/GUI/Batch/BatchPresenter.cpp

```diff
@@ -94,9 +94,15 @@ void BatchPresenter::notifyRestoreDefaultsRequested() {
 
 void BatchPresenter::notifySettingsChanged() { settingsChanged(); }
 
-void BatchPresenter::notifyReductionResumed() { resumeReduction(); }
+void BatchPresenter::notifyReductionResumed() {
+  m_mainPresenter->notifyProcessingResumed();
```

Comment:
The way widgets are normally enabled/disabled in this GUI is to update the processing state in the model and then do the notifications so that other presenters can update their own widgets' enabled/disabled accordingly. It would be good to keep this consistent. It might be easier to chat about this tomorrow, but to do this I think would go as follows:
- the model is already updated as the first step in `pauseReduction`
- the notification to the main window should be done after that
- the main window presenter could then have a single function (e.g. `updateWidgetEnabledState` to be consistent with other presenters) to enable/disable the menu options according to whether `isAnyBatchProcessing` or `isAnyBatchAutoreducing` is true

I think this will make us more flexible too. In future we might want to allow loading if we switch to another batch which is not currently processing, and this should allow that (although I'm not sure how easy it would be to identify which is the current batch, so I'd say that's out of scope for this PR).

Also, I'm not sure we need to disable saving, because I think it's only loading during processing that can cause problems? Probably not a big issue if you want to leave it for now, though.

Reply:
I have implemented a slightly different version from exactly what you suggested but it carries the principles, which will be available in the next commit.